src/conn/conn_sweep.c

```diff
@@ -45,6 +47,31 @@ __sweep_mark(WT_SESSION_IMPL *session, uint64_t now)
           __wt_atomic_loadi32(&dhandle->session_inuse) > 0 || dhandle->timeofdeath != 0)
             continue;
 
+        /*
+         * When MongoDB opens cursors on table URIs, WiredTiger opens and releases table dhandles,
```

Comment:
Hmm, I think this comment doesn't need the various symptoms of the issue such as contention etc. 
How about - 
"Retain the table dhandles for the corresponding active file dhandles. Accessing a file dhandle requires opening its associated table dhandle, so for any active file dhandle, a table dhandle must be present in the dhandle list to avoid repeated reopening."